File: sdk/ios/Sources/DuallerSDK/JSAdapter.swift
import UIKit

/// WebView adaptation layer for global JS objects
public final class JSAdapter {
    private let context: JSContext

    public init(context: JSContext) {
        self.context = context
    }

    public func injectGlobalObjects() {
        let globals = """
            __DUALLER__ = {
                version: "0.1.0",
                platform: "ios",
                engine: "javascriptcore"
            };
        """
        self.context.evaluateScript(globals)
    }

    public func postMessageToNative(_ message: String) {
        // Called from JS side
    }
}

File: sdk/android/src/main/java/com/dualler/sdk/adapter/JSAdapter.kt
package com.dualler.sdk.adapter

import com.dualler.sdk.view.DuallerWebView
import com.dualler.sdk.BuildConfig

/**
 * WebView adaptation layer -- handles JS <-> Native boundary
 */
class JSAdapter(private val webView: DuallerWebView) {

    fun injectGlobalObjects() {
        webView.evaluateJavascript("""
            window.__DUALLER__ = {
                version: '${BuildConfig.VERSION_NAME}',
                platform: 'android',
                engine: 'quickjs'
            };
        """.trimIndent(), null)
    }

    fun postMessageToNative(message: String) {
        // Called from JS side via AndroidBridge interface
    }
}

// Android side interface accessible from JS
interface AndroidBridgeInterface {
    fun handleMessage(message: String): String?
}
